pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -283,10 +285,27 @@ message ColumnOnUpdateExpression {
   Expression embedded_expr = 3 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
 
+// A reference between two relations, tracked down to index/column level.
+message Reference {
+  option (gogoproto.equal) = true;
+  // Both ends of this reference (relation `fromID` depends on relation `toID`).
+  uint32 fromID = 1 [
+    (gogoproto.customname) = "FromID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  uint32 toID = 2 [
+    (gogoproto.customname) = "ToID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  // If applicable, the ID of `toID` relation's index that is referenced by the `fromID` relation.
+  uint32 index_id = 3 [
+    (gogoproto.customname) = "IndexID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.IndexID"];
+  // The IDs of `toID` relation's columns that are referenced by the `fromID` relation.
+  repeated uint32 column_ids = 4 [(gogoproto.customname) = "ColumnIDs",
+    (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.ColumnID"];
```

Comment:
nit: title these as though they were go comments, i.e. PascalCase with the field name as it will be.